src/hooks/useHover/useHover.ts

```diff
@@ -1,9 +1,9 @@
 import { MutableRefObject, useEffect, useRef, useState } from "react";
 
 // Standard hook implementation https://usehooks.com/useHover/
-export default function useHover(): [MutableRefObject<HTMLElement>, boolean] {
+export default function useHover<T extends HTMLElement = HTMLElement>(): [MutableRefObject<T>, boolean] {
```

Comment:
is there a need in extending HTMLElement if your passing in a type and initializing it to be HTMLElement anyway?
BTW, I LOVE generics so I like it 😆

Reply:
There is a need actually, cause `T extends HTMLElement` means that whatever class is passed it should extend `HTMLElement`, which allows to use below some `HTMLElement` functions e.g. `addEventListener`
and = `HTLMElement` is just a default value for the the type